ui/pages/settings/experimental-tab/experimental-tab.component.js

```diff
@@ -71,16 +71,50 @@ export default class ExperimentalTab extends PureComponent {
     handleSettingsRefs(t, t('experimental'), this.settingsRefs);
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IN(blockaid)
+  toggleSecurityAlert(value) {
+    const { setSecurityAlertsEnabled, transactionSecurityCheckEnabled } =
+      this.props;
+    this.context.trackEvent({
+      category: MetaMetricsEventCategory.Settings,
+      event: 'Enabled/Disable security_alerts_enabled',
+      properties: {
+        action: 'Enabled/Disable security_alerts_enabled',
+        legacy_event: true,
+      },
+    });
+    setSecurityAlertsEnabled(!value);
+    if (!value && transactionSecurityCheckEnabled) {
+      this.toggleTransactionSecurityCheck(true);
+    }
+  }
+  ///: END:ONLY_INCLUDE_IN
+
+  toggleTransactionSecurityCheck(value) {
+    const { securityAlertsEnabled, setTransactionSecurityCheckEnabled } =
+      this.props;
+    this.context.trackEvent({
+      category: MetaMetricsEventCategory.Settings,
+      event: 'Enabled/Disable TransactionSecurityCheck',
+      properties: {
+        action: 'Enabled/Disable TransactionSecurityCheck',
+        legacy_event: true,
+      },
+    });
+    setTransactionSecurityCheckEnabled(!value);
+    if (!value && securityAlertsEnabled && this.toggleSecurityAlert) {
```

Comment:
If we are including the securityAlertsEnabled check here is it possible to run into a case with both settings toggled? Would the check be sufficient without this check?

Example scenario:
- securityAlertsEnabled is on
- user toggles Blockaid which toggles OpenSea off
- securityAlertsEnabled is off for some reason... 
- user toggles OpenSea
- securityAlertsEnabled is on
- Blockaid and OpenSea are toggled on

comment applies to both toggle functions

Reply:
I can not think of getting into this scenario.

Reply:
I think checking securityAlertsEnabled may be an additional check we don't need since we're also checking this.toggleSecurityAlert

Reply:
We need to toggle `securityAlertsEnabled` only if it is currently enabled.

Reply:
Ah right. I mixed this up with something else. Makes sense and that's how we could feel confident renaming the value param to currentValue or oldValue

Lgtm! Thanks for the updates :)